pkg/mapper/iac-providers/arm/types/template_test.go

```diff
@@ -0,0 +1,50 @@
+/*
+    Copyright (C) 2020 Accurics, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package types
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestIfTemplateStructIsInlineWithARMTemplate(t *testing.T) {
+	data, err := readTemplate("storage-account-create.json")
+	if err != nil {
+		t.Error(err)
+	}
+
+	var tmp Template
+	err = json.Unmarshal(data, &tmp)
```

Comment:
if possible, we should also verify the unmarshalled fields.

Reply:
Will do the changed similar to cft, in upcoming PR, because ti will also need changes in KaiMonkey